velox/type/Type.h

```diff
@@ -1119,6 +1119,58 @@ INTERVAL_DAY_TIME() {
   return IntervalDayTimeType::get();
 }
 
+constexpr long kMonthInYear = 12;
+/// Time interval in months.
+class IntervalYearMonthType : public IntegerType {
```

Comment:
Please also add a method Type::isIntervalYearMonth() similar to Type::isIntervalDayTime() https://github.com/facebookincubator/velox/pull/5244